pkg/engine/engine.go

```diff
@@ -257,6 +273,7 @@ func (e *Engine) DetectorAvgTime() map[string][]time.Duration {
 func (e *Engine) detectorWorker(ctx context.Context) {
 	for originalChunk := range e.chunks {
 		for chunk := range sources.Chunker(originalChunk) {
+			var chunkResults = make([]detectors.ResultWithMetadata, 0)
```

Comment:
**Nit (optional):** the default value of a slice (`nil`) can be `append`ed, so this is essentially equivalent:

```go
var chunkResults []detectors.ResultWithMetadata
```

Alternatively if you want to keep the `make`, I'd suggest the walrus operator instead of the `var` syntax:

```go
chunkResults := make([]detectors.ResultWithMetadata, 0)
```